src/Microsoft.Azure.SignalR.Common/Exceptions/AzureSignalRUnauthorizedException.cs

```diff
@@ -9,12 +9,16 @@ namespace Microsoft.Azure.SignalR.Common
     [Serializable]
     public class AzureSignalRUnauthorizedException : AzureSignalRException
     {
-        private const string ErrorMessage = "Authorization failed. Make sure you provide the correct connection string and have access to the resource.";
+        private const string ErrorMessage = "Authorization failed. Please check the connection string and role assignments. The role assignments will take up to 30 minutes to take effect if it was added recently.";
 
         public AzureSignalRUnauthorizedException(string requestUri, Exception innerException) : base(string.IsNullOrEmpty(requestUri) ? ErrorMessage : $"{ErrorMessage} Request Uri: {requestUri}", innerException)
         {
         }
 
+        internal AzureSignalRUnauthorizedException(Exception innerException) : base(ErrorMessage, innerException)
```

Comment:
Let's either use a more specific Aad exception or pass in the AAd exception message, instead of having aad specific error message inside this general exception class

Reply:
It's impossible.

First, we can't change an existing public exception name.
Second, it's impossible to determine if the a request was authorized by `AAD Token` or `AccessToken` when calling rest APIs.